src/core/server/ui_settings/saved_objects/permission_controlled_ui_settings_wrapper.ts

```diff
@@ -0,0 +1,172 @@
+    };
+
+    const updateUiSettingsWithPermission = async <T = unknown>(
+      type: string,
+      id: string,
+      attributes: Partial<T>,
+      options: SavedObjectsUpdateOptions = {}
+    ): Promise<SavedObjectsUpdateResponse<T>> => {
+      if (type === 'config' && id === DASHBOARD_ADMIN_SETTINGS_ID) {
+        try {
+          return ((await wrapperOptions.client.update<SavedObject<T>>(
+            type,
+            DASHBOARD_ADMIN_SETTINGS_ID,
+            attributes,
+            options
+          )) as unknown) as SavedObjectsUpdateResponse<T>;
+        } catch (error) {
+          if (SavedObjectsErrorHelpers.isNotFoundError(error)) {
+            return this.createPermissionUiSetting(attributes, options, wrapperOptions);
+          } else {
+            throw error;
+          }
+        }
+      }
+
+      return wrapperOptions.client.update<T>(type, id, attributes, options);
+    };
+
+    const bulkCreateWithPermission = async <T = unknown>(
+      objects: Array<SavedObjectsBulkCreateObject<T>>,
+      options?: SavedObjectsCreateOptions
+    ) => {
+      if (options?.id === DASHBOARD_ADMIN_SETTINGS_ID) {
+        throw new Error('Bulk create is not supported for admin settings');
+      }
+
+      // Check if any object is trying to create admin settings
+      const adminSettingObject = objects.find(
+        (obj) => obj.type === 'config' && obj.id === DASHBOARD_ADMIN_SETTINGS_ID
+      );
+
+      if (adminSettingObject) {
+        throw new Error('Bulk create is not supported for admin settings');
+      }
+
+      return wrapperOptions.client.bulkCreate<T>(objects, options);
+    };
+
+    const bulkUpdateWithPermission = async <T = unknown>(
+      objects: Array<SavedObjectsBulkUpdateObject<T>>,
+      options?: SavedObjectsBulkUpdateOptions
+    ) => {
+      const adminSettingObject = objects.find(
+        (obj) => obj.type === 'config' && obj.id === DASHBOARD_ADMIN_SETTINGS_ID
+      );
+
+      if (adminSettingObject) {
+        throw new Error('Bulk update is not supported for admin settings');
+      }
+
+      return wrapperOptions.client.bulkUpdate<T>(objects, options);
+    };
+
+    return {
+      ...wrapperOptions.client,
+      create: createUiSettingWithPermission,
+      bulkCreate: bulkCreateWithPermission,
+      delete: wrapperOptions.client.delete,
+      update: updateUiSettingsWithPermission,
+      bulkUpdate: bulkUpdateWithPermission,
+      get: wrapperOptions.client.get,
+      checkConflicts: wrapperOptions.client.checkConflicts,
+      errors: wrapperOptions.client.errors,
+      addToNamespaces: wrapperOptions.client.addToNamespaces,
+      deleteFromNamespaces: wrapperOptions.client.deleteFromNamespaces,
+      find: wrapperOptions.client.find,
+      bulkGet: wrapperOptions.client.bulkGet,
+      deleteByWorkspace: wrapperOptions.client.deleteByWorkspace,
+    };
+  };
+
+  private checkAdminPermission(wrapperOptions: SavedObjectsClientWrapperOptions): void {
```

Comment:
I think better to rename it something like `hasUISettingsWritePermission`

Reply:
Done

Reply:
My bad, actually `checkUISettingsWritePermission` is more appropriate.